### PR TITLE
feat: new dart codegen

### PR DESCRIPTION
**Pull request by laiiihz**
* generate dart code using [code_builder](https://pub.dev/packages/code_builder) & [dart_style](https://pub.dev/packages/dart_style)
* add dart [dio](https://pub.dev/packages/dio) code gen support
*  rewrite dart http code gen
* add dart dio code gen tests
* update dart http code gen tests(format dart code before compare code)

### CODE REVIEW

**Comment by ashitaprasad** [comment]
Thank you for submitting the PR @laiiihz 
I will review it and add my comments.

File: test/codegen/dart_http_codegen_test.dart [hunk 1]
```diff
@@ -52,18 +63,18 @@ void main() async {
   }
 }
 """;
-      expect(dartHttpCodeGen.getCode(requestModelGet2, "https"), expectedCode);
+      expect(
+        dartHttpCodeGen.getCode(requestModelGet2, "https"),
+        expectedCode.format,
```
Comment (ashitaprasad) [inline test/codegen/dart_http_codegen_test.dart:68]: The `expectedCode` string should be exactly what is being generated by the codegen.
No programatic modification (`expectedCode.format`) must be done to it.

File: test/codegen/dart_dio_codegen_test.dart [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    print(s);
+  } catch (e, s) {
+    print(e);
+    print(s);
+  }
+}
+""";
+      expect(
+        dartDioCodeGen.getCode(requestModelGet1, "https"),
+        expectedCode.format,
```
Comment (ashitaprasad) [inline test/codegen/dart_dio_codegen_test.dart:39]: The `expectedCode` string should be exactly what is being generated by the codegen.
No programatic modification (`expectedCode.format`) must be done to it in any test case.

**Comment by ashitaprasad** [comment]
Hi @laiiihz, The PR looks good. Just some minor changes required as mentioned above. Thanks!

File: test/codegen/dart_http_codegen_test.dart (hunk 1 above)
Comment (laiiihz) [reply test/codegen/dart_http_codegen_test.dart]: > The expectedCode string should be exactly what is being generated by the codegen
Yes, you are right! I have added a commit to fix this.

**Comment by laiiihz** [comment]
@ashitaprasad Yes, you are right! I have added a commit to fix this.

**Comment by ashitaprasad** [comment]
LGTM 🚀
Thanks for contributing @laiiihz 🎉
Looking forward to more contributions from you. 🙂

**Pull request merged**